actors/market/src/lib.rs

```diff
@@ -482,6 +483,13 @@ impl Actor {
             .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
                 format!("failed to notify deal with proposal cid {}", valid_deal.cid)
             })?;
+
+            emit::deal_published(
+                rt,
+                valid_deal.proposal.client.id().unwrap(),
+                valid_deal.proposal.provider.id().unwrap(),
```

Comment:
Are you sure we guarantee that these are IDs? Especially the client?

I see here that we resolve the client:

https://github.com/filecoin-project/builtin-actors/blob/f8916da127502cb2dd24651466cd89a2ffbb1f38/actors/market/src/lib.rs#L291-L300

Reply:
@Stebalien Yeah this is fine because we do not add deal proposals with non-id clients to the `valid_deals` slice. See [here](https://github.com/filecoin-project/builtin-actors/blob/7e0abcacc4ad8c1d1ff4ef404f9f173380388acf/actors/market/src/lib.rs#L296) AND [here](https://github.com/filecoin-project/builtin-actors/blob/7e0abcacc4ad8c1d1ff4ef404f9f173380388acf/actors/market/src/lib.rs#L383).

The `valid_deals` slice is then used to emit these events.

Reply:
Ah, the relevent code is https://github.com/filecoin-project/builtin-actors/blob/7e0abcacc4ad8c1d1ff4ef404f9f173380388acf/actors/market/src/lib.rs#L336-L337